internal/app/release.go

```diff
@@ -172,79 +194,74 @@ func validateReleaseCharts(s *state) error {
+					"apps [" + apps + "] but the chart found at that path has version [ " + v + " ] which does not match."
 				return
 			}
-			matches := versionExtractor.FindStringSubmatch(result.output)
-			if len(matches) == 2 {
-				version := strings.Trim(matches[1], `'"`)
-				if strings.Trim(r.Version, `'"`) != version {
-					c <- "Chart [ " + r.Chart + " ] with version [ " + r.Version + " ] is specified for " +
-						"app [" + app + "] but the chart found at that path has version [ " + version + " ] which does not match."
-					return
-				}
-			}
-
-		} else {
-			version := r.Version
-			if len(version) == 0 {
-				version = "*"
-			}
-			cmd := helmCmd([]string{"search", "repo", r.Chart, "--version", version, "-l"}, "Validating [ "+r.Chart+" ] chart's version [ "+r.Version+" ] availability")
+		}
+	} else {
+		v := version
+		if len(v) == 0 {
+			v = "*"
+		}
+		cmd := helmCmd([]string{"search", "repo", chart, "--version", v, "-l"}, "Validating [ "+chart+" ] chart's version [ "+version+" ] availability")
 
-			if result := cmd.exec(); result.code != 0 || strings.Contains(result.output, "No results found") {
-				c <- "Chart [ " + r.Chart + " ] with version [ " + r.Version + " ] is specified for " +
-					"app [" + app + "] but was not found. If this is not a local chart, define its helm repo in the helmRepo stanza in your DSF."
-				return
-			}
+		if result := cmd.exec(); result.code != 0 || strings.Contains(result.output, "No results found") {
+			c <- "Chart [ " + chart + " ] with version [ " + version + " ] is specified for " +
+				"apps [" + apps + "] but was not found. If this is not a local chart, define its helm repo in the helmRepo stanza in your DSF."
+			return
 		}
 	}
 }
 
 // getChartVersion fetches the lastest chart version matching the semantic versioning constraints.
 // If chart is local, returns the given release version
-func (r *release) getChartVersion() (string, string) {
-	if isLocalChart(r.Chart) {
-		return r.Version, ""
+func getChartVersion(chart, version string) (string, string) {
+	if isLocalChart(chart) {
+		log.Info("Chart [ " + chart + "] with version [ " + version + " ] was found locally.")
+		return version, ""
 	}
-	cmd := helmCmd([]string{"search", "repo", r.Chart, "--version", r.Version, "-o", "json"}, "Getting latest chart's version "+r.Chart+"-"+r.Version+"")
+
+	cmd := helmCmd([]string{"search", "repo", chart, "--version", version, "-o", "json"}, "Getting latest non-local chart's version "+chart+"-"+version+"")
 
 	result := cmd.exec()
 	if result.code != 0 {
-		return "", "Chart [ " + r.Chart + " ] with version [ " + r.Version + " ] is specified but not found in the helm repositories"
+		return "", "Chart [ " + chart + " ] with version [ " + version + " ] is specified but not found in the helm repositories"
 	}
 
-	chartVersions := make([]chartVersion, 0)
-	if err := json.Unmarshal([]byte(result.output), &chartVersions); err != nil {
+	cv := make([]chartVersion, 0)
```

Comment:
out of curiosity - why are you changing more comprehensive variable names to simply not intuitive letters?

Reply:
fixed this